sed/dataset/dataset.py

```diff
@@ -0,0 +1,285 @@
+"""This module provides functions to fetch and load datasets."""
+from __future__ import annotations
+
+import os
+import shutil
+import zipfile
+from pathlib import Path
+
+import requests
+
+from sed.core.config import load_config
+from sed.core.config import save_config
+from sed.core.logging import setup_logging
+from sed.core.user_dirs import construct_module_dirs
+
+# Configure logging
+logger = setup_logging(__name__)
+
+NAME = "datasets"
+user_paths = construct_module_dirs(NAME)
+json_path_user = user_paths["config"].joinpath(NAME).with_suffix(".json")
+json_path_module = Path(os.path.dirname(__file__)).joinpath(NAME).with_suffix(".json")
+
+
+def load_datasets_dict() -> dict:
+    """
+    Loads the datasets configuration dictionary from the user's datasets JSON file.
+
+    If the file does not exist, it copies the default datasets JSON file from the module
+    directory to the user's datasets directory.
+
+    Returns:
+        dict: The datasets dict loaded from user's datasets JSON file.
+    """
+    # check if datasets.json exists in user_config_dir
+    if not os.path.exists(json_path_user):
+        shutil.copy(json_path_module, json_path_user)
+    datasets = load_config(str(json_path_user))
+    return datasets
+
+
+def available_datasets() -> list:
+    """
+    Returns a list of available datasets.
+
+    Returns:
+        list: List of available datasets.
+    """
+    datasets = load_datasets_dict()
+    return list(datasets)
```

Comment:
Wouldn't it make sense to put all this into a class, and make this an attribute?

Reply:
Yes. So you suggest the user first creates a class instance and then call class.load_dataset() etc?

Reply:
It's true this will make it somewhat more difficult to use. Up for discussion.